doc/src/extras.rst

```diff
@@ -141,6 +141,262 @@ Logging cursor
+                         logical replication)
+       :param writer: a file-like object to write replication messages to
+       :param start_lsn: the LSN position to start from, in the form
+                         ``XXX/XXX`` (forward-slash separated pair of
+                         hexadecimals)
+       :param timeline: WAL history timeline to start streaming from (optional,
+                        can only be used with physical replication)
+       :param keepalive_interval: interval (in seconds) to send keepalive
+                                  messages to the server
+       :param options: a dictionary of options to pass to logical replication
+                       slot
+
+       When used on non-asynchronous connection this method enters an endless
+       loop, reading messages from the server and passing them to ``write()``
+       method of the *writer* object.  This is similar to operation of the
+       `~cursor.copy_to()` method.  It also sends keepalive messages to the
+       server, in case there were no new data from it for the duration of
+       *keepalive_interval* seconds (this parameter's value must be equal to
+       at least than 1 second, but it can have a fractional part).
+
+       With asynchronous connection, this method returns immediately and the
+       calling code can start reading the replication messages in a loop.
+
+       A sketch implementation of the *writer* object for logical replication
+       might look similar to the following::
+
+           from io import TextIOBase
+
+           class LogicalStreamWriter(TextIOBase):
+
+               def write(self, msg):
+                   self.store_message_data(msg.payload)
+
+                   if self.should_report_to_the_server_now(msg):
+                       msg.cursor.send_replication_feedback(flush_lsn=msg.wal_end)
+
+       First, like with the `~cursor.copy_to()` method, the code that calls
+       the provided ``write()`` method checks if the *writer* object is
+       inherited from `~io.TextIOBase`.  If that is the case, the message
+       payload to be passed is converted to unicode using the connection's
+       `~connection.encoding` information.  Otherwise, the message is passed
+       as is.
+
+       The *msg* object being passed is an instance of `~ReplicationMessage`
+       class.
+
+       After storing certain amount of messages' data reliably, the client
+       should send a confirmation message to the server.  This should be done
+       by calling `~send_replication_feedback()` method on the corresponding
+       replication cursor.  A reference to the cursor is provided in the
+       `~ReplicationMessage` as an attribute.
+
+       .. warning::
+
+           Failure to properly notify the server by constantly consuming and
+           reporting success at appropriate times can eventually lead to "disk
+           full" condition on the server, because the server retains all the
```

Comment:
Disk full is only a risk when using a replication slot, fwiw. Physical replication without a slot can't cause disk full, and is one of the reasons it remains useful.